pydis_site/apps/api/viewsets/bot/bumped_thread.py

```diff
@@ -0,0 +1,65 @@
+from rest_framework.mixins import (
+    CreateModelMixin, DestroyModelMixin, ListModelMixin, RetrieveModelMixin
+)
+from rest_framework.viewsets import GenericViewSet
+
+from pydis_site.apps.api.models.bot import BumpedThread
+from pydis_site.apps.api.serializers import BumpedThreadSerializer
+
+
+class BumpedThreadViewSet(
+    GenericViewSet, CreateModelMixin, DestroyModelMixin, RetrieveModelMixin, ListModelMixin
+):
+    """
+    View providing CRUD (Minus the U) operations on threads to be bumped.
+
+    ## Routes
+    ### GET /bot/bumped-threads
+    Returns all BumpedThread items in the database.
+
+    #### Response format
+    >>> [
+    ...     {
+    ...         'thread_id': "941705627405811793",
+    ...     },
+    ...     ...
+    ... ]
+
+    #### Status codes
+    - 200: returned on success
+    - 401: returned if unauthenticated
+
+    ### GET /bot/bumped-threads/<thread_id:int>
+    Returns a specific BumpedThread item from the database.
+
+    #### Response format
+    >>> {
+    ...     'thread_id': "941705627405811793",
+    ... }
+
+    #### Status codes
+    - 200: returned on success
+    - 404: returned if a BumpedThread with the given thread_id was not found.
+
+    ### POST /bot/bumped-threads
+    Adds a single BumpedThread item to the database.
+
+    #### Request body
+    >>> {
+    ...    'thread_id': int,
+    ... }
```

Comment:
Similar to the GET request, it should be possible to just directly send the integer value. Hell, we could even avoid JSON and do url-encoded params or maybe `POST /bot/bumped-threads/<thread_id:int>` if DRF supports that for post requests and it's not deemed unidiomatic.

Reply:
https://www.django-rest-framework.org/api-guide/routers/#simplerouter suggests that POST isn't supported with a lookup path `POST /bot/bumped-threads/<thread_id:int>` only PUT or PATCH would be.

However, query params would work. I'm not sure how much better this is over just using request body though since it would be adding `?thread_id=int` to the url instead of adding `{thread_id: int}` to the request body

Reply:
Never mind then. In practice, it doesn't matter since for aiohttp a dictionary needs to be passed either way.